### ISSUE TITLE
[BUG] Wrong Export of the KITTI dataset to YOLOv5 format

### ISSUE THREAD

**Issue by DJE98** [voxel51/fiftyone#5520]
### Describe the problem

Fiftyone does not export a compliant version of the [KITTI dataset](https://docs.voxel51.com/dataset_zoo/datasets.html#dataset-zoo-kitti) in [yolov5 format](https://docs.voxel51.com/user_guide/export_datasets.html#yolov5dataset-export) when train and test split should be exported. I orientated my code on the [ultralytics fiftyone docs](https://docs.voxel51.com/integrations/ultralytics.html). The exported dataset should have the following format by [ultralytics yolo](https://docs.ultralytics.com/datasets/#oriented-bounding-boxes-obb):

dataset/
├──dataset.yaml
├── train/
│   ├── images/
│   └── labels/
└── val/
    ├── images/
    └── labels/

However the output produces the following output without val labels:

dataset/
├──dataset.yaml
├── images/
│   ├── train/
│   └── val/
└── labels/
    └── train/

The test dataset from KITTI has to be exported as val. This is a requirement of the format. The test labels also won't be export, if use test as export name. As result of the wrong folder structure no labels can be found.

### Code to reproduce issue

```Python
import fiftyone as fo
import fiftyone.zoo as foz

classes = ["Pedestrian", "Truck", "Car", "Cyclist", "DontCare", "Misc", "Van", "Tram", "Person_sitting"]

fo.config.database_uri = "mongodb://localhost:27017"

train_dataset = foz.load_zoo_dataset(
    "kitti",
    split="train",
)
train_dataset.export(
    export_dir="kitti_yolo",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="ground_truth",
    split="train",
    classes=classes,
)

validation_dataset = foz.load_zoo_dataset(
    "kitti",
    split="test",
)
validation_dataset.export(
    export_dir="kitti_yolo",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="ground_truth",
    split="val",
    classes=classes,
    )
```

### System information

-   **OS Platform and Distribution**: NixOS 24.05
-   **Python version** 3.10.16:
-   **FiftyOne version** : 1.3.0
-   **FiftyOne installed from** (pip or source): pip

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community
-   [x] No. I cannot contribute a bug fix at this time

**Comment by brimoor** [comment]
Hmm that's strange... I wonder if Ultralytics changed their recommended format, because the format that we implemented was specifically chosen to match what Ultralytics expected at the time (4+ years ago).

I agree that [this Ultralytics page](https://docs.ultralytics.com/datasets/#oriented-bounding-boxes-obb) seems to want data in this format:

```
dataset/
├── train/
│   ├── images/
│   └── labels/
└── val/
    ├── images/
    └── labels/
```

However, the [YOLOv5 format we currently export](https://docs.voxel51.com/user_guide/export_datasets.html#yolov5dataset) is this:

```
dataset/
    dataset.yaml
    images/
        train/
        val/
    labels/
        train/
        val/
```

Fortunately, our YOLOv5 exporter is flexible enough to export in the new format. You can achieve this by passing the `data_path` and `labels_path` arguments like so:

```py
import fiftyone as fo
import fiftyone.zoo as foz
import fiftyone.utils.random as four

dataset = foz.load_zoo_dataset("quickstart")

dataset.untag_samples(dataset.distinct("tags"))
four.random_split(dataset, {"train": 0.7, "test": 0.2, "val": 0.1})

# Test split has no labels
dataset.match_tags("test").clear_sample_field("ground_truth")

classes = dataset.distinct("ground_truth.detections.label")

for split in dataset.distinct("tags"):
    dataset.match_tags(split).export(
        export_dir="/tmp/yolov5",
        data_path=f"{split}/images",
        labels_path=f"{split}/labels",
        dataset_type=fo.types.YOLOv5Dataset,
        label_field="ground_truth",
        split=split,
        classes=classes,
    )
 ```

**Comment by brimoor** [comment]
@DJE98 do you happen to know if Ultralytics changed their format?

We can easily update our default YOLOv5 export format to match [this](https://docs.ultralytics.com/datasets/#oriented-bounding-boxes-obb), I'd just like to know whether we need to support the old format too, or what exactly is going on 🤔

**Comment by DJE98** [comment]
I think that a backwards compatibility ist not necessary, because you also can run the [old yolo versions ](https://docs.ultralytics.com/models/yolo11/) with the new library. However I could not find the exact ultalytics yolo version, which changes the format. Why is there no label output for the val dataset?

**Comment by manushreegangwar** [comment]
@DJE98 I trained a YOLO model using [FiftyOne Ultralytics export in YOLOv5 format](https://docs.voxel51.com/integrations/ultralytics.html#training). The training works for me. See code below:

```
import fiftyone as fo
from ultralytics import YOLO

dataset = fo.load_dataset("my_dataset")
train = dataset.match_tags("train")
val = dataset.match_tags("validation")
classes = [...]

train.export(
    export_dir="path/to/export_dir",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="ground_truth",
    split="train",
    classes=classes,
)

val_view.export(
    export_dir="path/to/export_dir",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="ground_truth",
    split="val",
    classes=classes,
)

dataset_yaml = "path/to/export_dir/dataset.yaml"
model = YOLO("yolo11n.pt")
train_results = model.train(data=dataset_yaml)
val_results = model.val()
```

The `dataset.yaml` file generated by FiftyOne YOLOv5 exporter has the following format:

```
names:
  0: class_0
  1: class_1
  ...
path: path/to/export_dir
train: ./images/train/
val: ./images/val/
```

and the export directory has the following structure:

```
dataset/
├──dataset.yaml
├── images/
│ ├── train/
│ └── val/
└── labels/
│ ├── train/
│ └── val/
```

The above directory structure seems to be in accordance with [Ultralytics YOLO format](https://docs.ultralytics.com/datasets/detect/). However, there seems to be a discrepancy in the [Ultralytics page that you shared](https://docs.ultralytics.com/datasets/#steps-to-contribute-a-new-dataset).

A couple of questions:
- Have you tried using datasets other than `KITTI`? Do you run into the same issue? 
- Have you verified that `validation_dataset` samples have the expected "ground_truth" labels?

```
validation_dataset = foz.load_zoo_dataset(
    "kitti",
    split="test",
)
```

**Comment by brimoor** [comment]
> Why is there no label output for the val dataset? 

The KITTI test split is unlabeled, so that's why you're not seeing labels. FiftyOne's `export()` method will always export labels if they exist on the dataset.

**Comment by DJE98** [comment]
@brimoor If I set the path explicitly, the kitti dataset will be exported correctly. Thanks for the tip.
@manushreegangwar Which dataset do you use?

**Comment by manushreegangwar** [comment]
@DJE98 I used `BDD100k`
